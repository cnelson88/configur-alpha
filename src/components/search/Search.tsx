import React, { useEffect, useState } from 'react';
import SearchResults from './SearchResults';
import DefaultData from './DefaultData';

import { Input } from '../styles/input';

const Search = () => {
    const [searchQuery, setSearchQuery] = useState<string>('');    
    const [queryResults, setQueryResults] = useState<Array<string | number>>([]);

    useEffect(() => {
        setSearchQuery(searchQuery);              
    });

    useEffect(() => {
            if (searchQuery.length > 0) {
                const apiUrl = 'https://jsonplaceholder.typicode.com/users';
                
                fetch(apiUrl)
                    .then((response) => response.json())
                    .then((response) => {
                        response.filter((value: any) => {
                            if(value.name.includes(searchQuery)) {
                                return setQueryResults([ ...queryResults, value ]);
                            } else if (value.username.includes(searchQuery)) {
                                return setQueryResults([ ...queryResults, value ]);
                            }
                        })    
                    });
            }
    }, [searchQuery]);

    const searchInput = () => {
        return (
            <>
                <Input
                    type='text'
                    placeholder='Please enter a name...'
                    value={searchQuery}
                    onChange={handleChange}
                    onFocus={clearInput}
                />       
            </>
        )
    };

    const handleChange = (e: any) => {
        const casing = e.target.value;
        const casingUpper = casing.charAt(0).toUpperCase() + casing.slice(1);

        setSearchQuery(casingUpper);
        setQueryResults([]);
    };

    const clearInput = () => {
        setSearchQuery('');
    }

    return (
        <>
            {searchInput()}
            {queryResults.length > 0 ? <SearchResults queryResults={queryResults} /> : <DefaultData />}
        </>
    );
};

export default Search;