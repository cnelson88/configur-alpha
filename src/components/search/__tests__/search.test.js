import React from 'react';
import { shallow } from 'enzyme';

import Search from '../search.jsx';

describe('Search', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Search debug />);

        expect(component).toMatchSnapshot();
    });

    it('should render correctly with no props', () => {
        const component = shallow(<Search />);
    
        expect(component).toMatchSnapshot();
    });
});
