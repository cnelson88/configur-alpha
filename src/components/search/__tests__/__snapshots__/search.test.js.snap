// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly in "debug" mode 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    placeholder="Please enter a name..."
    type="text"
    value=""
  />
  <p>
    No results found.
  </p>
</Fragment>
`;

exports[`Search should render correctly with no props 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    placeholder="Please enter a name..."
    type="text"
    value=""
  />
  <p>
    No results found.
  </p>
</Fragment>
`;

exports[`should render correctly with no props 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    placeholder="Please enter a name..."
    type="text"
    value=""
  />
  <p>
    No results found.
  </p>
</Fragment>
`;
